app/javascript/controllers/complex_checkbox_controller.js

```diff
@@ -0,0 +1,46 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = ["checkbox", "checkedSection", "uncheckedSection"]
+
+  connect() {
+    this.toggleInputs()
+  }
+
+  toggleInputs(event) {
+    if(this.checkboxTarget.checked) {
+      this.toggle(this.checkedSectionTarget, this.uncheckedSectionTarget, !!event)
+    } else {
+      this.toggle(this.uncheckedSectionTarget, this.checkedSectionTarget, !!event)
+    }
+  }
+
+  toggle(target, source, copy) {
+    target.hidden = false
+    source.hidden = true
+
+    const targetInputs = target.querySelectorAll('select,input[type="number"],input[type="checkbox"]')
```

Comment:
Can the data copying be extracted to a separate method?

Reply:
Done.